internal/application.go

```diff
@@ -88,35 +155,42 @@ func (app *Application) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	for prefix := range clientsMapping {
 		if strings.HasPrefix(path, prefix) {
-			if common.IsSecurityEnabled() {
-				app.secure(w, r)
-				return
-			}
 			originalPath := strings.TrimPrefix(path, prefix)
 			targetAddr := clientsMapping[prefix]
-			insecure(w, r, originalPath, targetAddr)
+			if common.IsSecurityEnabled() {
+				app.secure(w, r, originalPath, targetAddr)
+			} else {
+				insecure(w, r, originalPath, targetAddr)
+			}
+			return
 		}
 	}
 }
 
-func (app *Application) secure(w http.ResponseWriter, r *http.Request) {
-	targetHost := fmt.Sprintf("%s:%d", app.config.APIGateway.Server, app.config.APIGateway.ApplicationPort)
+func (app *Application) secure(w http.ResponseWriter, r *http.Request, originalPath string, client Client) {
+	defer r.Body.Close()
+	origin, _ := url.Parse(client.addr)
```

Comment:
Don't ignore errors. Should panic is error occurs. Yes I said panic since not good way to pass error back and exit and we know this is during start-up

Reply:
I kept this as it was before (see old line 113) but I can add a panic here if an error happens.